### ISSUE TITLE
debuginfo-gdb tests can't handle new gdb that supports rust

### ISSUE THREAD

**Issue by infinity0** [rust-lang/rust#36323]
Lots of `error: line not found in debugger output` e.g. see [buildlog for 1.13.0~~nightly.20160904](https://launchpadlibrarian.net/282731263/buildlog_ubuntu-yakkety-amd64.rustc_1.13.0~~nightly.20160904+dfsg1-1_BUILDING.txt.gz), shorter snippet of one failure: https://paste.debian.net/812883/

```
test result: FAILED. 37 passed; 65 failed; 3 ignored; 0 measured
```

**Comment by michaelwoerister** [comment]
@brson @alexcrichton We need test infrastructure that uses a Rust-enabled version of GDB.

cc @tromey @Manishearth

**Comment by Manishearth** [comment]
`set language rust` might be necessary in those tests?

**Comment by alexcrichton** [comment]
Discussed during tools triage today, conclusions were:

* Ideally tests would support all versions of GDB, perhaps even including a gdb/lldb style split of assertions and commands for newer versions of GDB
* We should have bots testing both versions of GDB
  * This may involve two bots with two versions of GDB
  * It may also involve one bot testing two versions of GDB

**Comment by nagisa** [comment]
This is an example of a failure, which also makes it quite obvious *why* the tests fail:

```
error: line not found in debugger output: $4 = {a = 3, b = 4, c = 5, d = 6, e = 7, f = 8, g = 9, h = 10}
<snip>
$4 = function_arg_initialization::BigStruct {a: 3, b: 4, c: 5, d: 6, e: 7, f: 8, g: 9, h: 10}
```

This is a very nasty failure case because I haven’t seen a successful run of tests on arch for months now.

**Comment by TimNN** [comment]
I've got a PR ready, with fixes all those tests (for GDB trunk), #37410.

**Comment by infinity0** [comment]
#37410 was closed, does that mean this is now fixed? It doesn't seem to be in 1.14.0, will it be in 1.15.0?

**Comment by infinity0** [comment]
Ping! Can this be closed? A fix was apparently merged in #37597

**Comment by michaelwoerister** [comment]
Yes, I think this can be considered fixed.
